packages/devtools-map-bindings/src/utils.js

```diff
@@ -1,35 +1,33 @@
 // @flow
 
 import type { MappedScopeBindings } from "debugger-html";
+const { default: generate } = require("babel-generator");
+
+const babylon = require("babylon");
+const { default: traverse } = require("babel-traverse");
 
 function replaceOriginalVariableName(
   expression: string,
   generatedScopes: MappedScopeBindings[]
 ): string {
-  // JavaScript indetifier is a complex thing: it can contain '$', '\', ZWJ, any
-  // unicode letter or number. Simplifing that to latin character set for now.
-  // For simplicity, search only first characters of the expression to find an
-  // identifier/variable name.
-  const possibleVarNameRegex = /^([$\w]+)/;
-  const matchedVarName = possibleVarNameRegex.exec(expression);
-  if (!matchedVarName) {
-    return expression;
-  }
+  const ast = babylon.parse(expression, {
```

Comment:
babylon has the parseExpression method, which returns Expression tree (not Script)